xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp

```diff
@@ -123,6 +123,19 @@ bool CActiveAEBufferPool::Create(unsigned int totaltime)
   config.sample_rate = m_format.m_sampleRate;
   config.channel_layout = CAEUtil::GetAVChannelLayout(m_format.m_channelLayout);
 
+  // discard already created buffers
+  if (m_allSamples.size() > 0 || m_freeSamples.size() > 0)
```

Comment:
I kniw that it is not right, but is needed because adsp can change his format according to active or non avtive add-ons. The main problem is that AE doesn't get any response about changes from adsp. Consequently I jave to recreate the buffers internally.